### PR TITLE
Yocto recipe for Kanto Suite Bootstrapping

### PR DESCRIPTION
**Pull request by hristobojilov**
[#3] Yocto recipe for Kanto Suite Bootstrapping

Signed-off-by: Hristo Bozhilov <Hristo.Bozhilov@bosch.io>

### CODE REVIEW

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# ensure all additional resources are properly packed in the resulting package if provided
+FILES:${PN} += "${SB_CFG_DD}/suite-bootstrapping/config.json"
+FILES:${PN} += "${SB_CFG_DD}/suite-bootstrapping/provisioning.json"
+FILES:${PN} += "${SB_CFG_DD}/suite-bootstrapping/iothub.crt"
+
+
+RDEPENDS:${PN} += "mosquitto"
+
+PROVIDES:${PN} += "edge/suite-bootstrapping"
+RPROVIDES:${PN} += "edge/suite-bootstrapping"
```
Comment (konstantina-gramatova) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb:41]: Must be prefixed with `kanto` not `edge`.

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb [hunk 2]
```diff
@@ -0,0 +1,79 @@
+DESCRIPTION = "Eclipse Kanto - Suite Bootstrapping Agent"
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb:1]: Remove "Agent" from the description.

File: recipes-connectivity/suite-bootstrapping/files/service.template [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+Requires=mosquitto.service
+
+[Service]
+Type=simple
+ExecStart=@SB_BIN_DD@/suite-bootstrapping -configFile @SB_CFG_DD@/suite-bootstrapping/config.json
+Restart=always
+
+[Install]
+WantedBy=multi-user.target
+
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/files/service.template:15]: No need to have two empty lines. Only one should be enough.

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping.inc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Common configuration variables to set the binaries and resources allocation directories for Eclipse Kanto suite bootstrapping component
+# The destination directory where the required configuration files for the proper connectivity establishment of the bootstrapping agent will be placed
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping.inc:2]: "bootstrapping agent" should be replaced by: suite bootstrapping

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping.inc [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+SB_CFG_DD ?= "${sysconfdir}"
+# The destination directory where the generated service file will be placed
+SB_SYSUNIT_DD ?= "${systemd_unitdir}/system"
+# The destination directory where the suite bootstrapping agent binaries will be placed
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping.inc:6]: Remove "agent".

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping.inc [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+SB_BIN_DD ?= "${bindir}"
+# The destination directory where the suite bootstrapping agent logs will be placed
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping.inc:8]: Remove "agent".

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        -e 's,@SB_LOG_DD@,${SB_LOG_DD},g' \
+        -i ${D}${SB_CFG_DD}/suite-bootstrapping/config.json
+  fi
+  
+  # provisioning.json
+  if  [ -f "${THISDIR}/files/provisioning.json" ]; then
+    install -m 0644  ${THISDIR}/files/provisioning.json ${D}${SB_CFG_DD}/suite-bootstrapping
+  fi
+}
+
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb:79]: No need to use two empty lines.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+PROVIDES:${PN} += "kanto/suite-bootstrapping"
+RPROVIDES:${PN} += "kanto/suite-bootstrapping"
+
+do_install() {
+  install -d "${D}/${SB_BIN_DD}"
+  
+  install -m 0755 "${GO_BUILD_BINDIR}/bootstrapping" "${D}${SB_BIN_DD}/suite-bootstrapping"
+  
+  if ${@bb.utils.contains('DISTRO_FEATURES','systemd','true','false',d)}; then
+   install -d ${D}${SB_SYSUNIT_DD}
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb:49]: The formatting is broken here, one space is missing.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+do_compile[network] = "1"
+
+FILES:${PN} += "${SB_SYSUNIT_DD}/suite-bootstrapping.service"
+FILES:${PN} += "${SB_BIN_DD}/suite-bootstrapping"
+# ensure all additional resources are properly packed in the resulting package if provided
+FILES:${PN} += "${SB_CFG_DD}/suite-bootstrapping/config.json"
+FILES:${PN} += "${SB_CFG_DD}/suite-bootstrapping/provisioning.json"
+FILES:${PN} += "${SB_CFG_DD}/suite-bootstrapping/iothub.crt"
+
+
```
Comment (e-grigorov) [inline recipes-connectivity/suite-bootstrapping/suite-bootstrapping_git.bb:37]: We don't need two empty lines here, only one is enough.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Review by konstantina-gramatova — APPROVED** [review]

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**